trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/plugins/generator/cacsl2boogietranslator/witness/GraphMLExtractedCorrectnessWitness.java

```diff
@@ -11,11 +38,17 @@
 import de.uni_freiburg.informatik.ultimate.util.datastructures.ImmutableSet;
 import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.Pair;
 
+/**
+ * Stores the invariants extracted from a correctness witness in GraphML. To ensure that each invariant is only
+ * instrumented once, store the labels of the GraphML nodes alongside.
+ *
+ * @author Frank Schüssele (schuessf@informatik.uni-freiburg.de)
+ */
 public class GraphMLExtractedCorrectnessWitness implements IExtractedCorrectnessWitness {
 	private final Map<IASTNode, Pair<ImmutableSet<String>, ExtractedWitnessInvariant>> mInvariants;
 	private final Set<ImmutableSet<String>> mNodeLabelsOfAddedWitnesses = new HashSet<>();
 
-	public GraphMLExtractedCorrectnessWitness(
+	GraphMLExtractedCorrectnessWitness(
```

Comment:
Do you omit the access modifier to want the constructor and methods to be package-private?
If you don't want to use them outside of the package in the future, then that's a good decision.

Reply:
Yes, the constructor should be only accessible during construction by `GraphMLCorrectnessWitnessExtractor`, only the getters should be public. So maybe it would be even better to make this implementation a nested class in `GraphMLCorrectnessWitnessExtractor` (as it is also rather small and simple).

Reply:
Sure, you could implement it as a nested class, but I think you can leave it as it is.